test/migration.spec.ts

```
@@ -220,20 +292,21 @@ describe('migration', () => {
     });
 
     it('should fail with an error message if the migration is invalid', () => {
-      const invalidMigrationName = 'invalid-migration';
-
-      const migration = new Migration(
-        dbMock,
-        invalidMigrationName,
-        {},
-        options,
-        {},
-        logger
-      );
-
       const direction = 'up';
+      const invalidMigrationName = 'invalid-migration';
 
-      expect(() => migration.apply(direction)).toThrow(
```

**question:** is this change required? or can it be reverted?

---

Everything can be reverted :)

I think, though, that the current behavior leads to unexpected side effects and wrong error messages, since the test case `should fail with an error message if the migration is invalid` is expected to throw an error message starting with `Unknown value for direction` which is misleading ([rows 294...314](https://github.com/salsita/node-pg-migrate/blob/c9fefd2ec8bb4f4e066f6eeba860c725021ec997/test/migration.spec.ts#L294-L314)).

---

Maybe you misunderstood me 😅
I just meant in this test-case that the construct call should not be part of the expect, but only the apply-call.